### DISCUSSION TITLE
Twig/Template cache into Redis

### DISCUSSION THREAD

**Discussion by eerison** [symfony/symfony#48674] [Q&A] [answered]
Hello I`m trying to save twig cache into redis, But I didnt see anyway to make it works :/
the only thing that I found was pass the path like here: https://symfony.com/doc/current/reference/configuration/twig.html#cache, But it doesn't help me because it saves locally!

is there someway to use like [cache adapter](https://symfony.com/doc/current/components/cache.html#available-cache-adapters)?

**Comment by norkunas** [comment]
If your app cache adapter is configured as redis, then you can use `{% cache %}` tag for that

**Reply by eerison** [reply]
I don't wan't to save one specific part, I want to save my whole twig cache!

The reason is: I'm using serverless and I cant save it locally!

**Comment by MatTheCat** [comment, marked as answer]
Why do you want to use Redis? From [Twig’s source code](https://github.com/twigphp/Twig/blob/e5c87f882183134a0e56845fd3e2e63c4a10396a/src/Cache/CacheInterface.php#L17-L19):

> It is highly recommended to always store templates on the filesystem to benefit from the PHP opcode cache.

So you could change the `twig` service definition to use a custom `CacheInterface`, but I would advise against it.

**Reply by eerison** [reply]
> Why do you want to use Redis? From [Twig’s source code](https://github.com/twigphp/Twig/blob/e5c87f882183134a0e56845fd3e2e63c4a10396a/src/Cache/CacheInterface.php#L17-L19):
> 
> > It is highly recommended to always store templates on the filesystem to benefit from the PHP opcode cache.
> 
> So you could change the `twig` service definition to use a custom `CacheInterface`, but I would advise against it.

Hey @MatTheCat 

> Why do you want to use Redis?

As I am using serverless I cann't save anything into the function and I don't want to lost my twig/framework cache

> So you could change the twig service definition to use a custom

hmm do you have some example?

**Reply by eerison** [reply]
Ok if it's hight recommend store in [filesystem](https://github.com/twigphp/Twig/blob/e5c87f882183134a0e56845fd3e2e63c4a10396a/src/Cache/CacheInterface.php#L17-L19), should I save like in my bucket? if yes how should I do this for cache 👀

**Reply by stof** [reply]
I would say that you should probably warmup the Twig cache (by compiling all your templates) during your deployment, so that the cache is part of the image that you deploy in your serverless runtime.

**Reply by eerison** [reply]
> I would say that you should probably warmup the Twig cache (by compiling all your templates) during your deployment, so that the cache is part of the image that you deploy in your serverless runtime.

hmmm it's a solution, but the problem is, my image gonna be so heavy :(

**Reply by eerison** [reply]
Maybe I should use the same approach for [cache:warmup](https://bref.sh/docs/frameworks/symfony.html#deploy), I will test it and see how big the image is!

**Reply by mbabker** [reply]
I'd worry less about the image size in this case and more about system reliability.  Deploying images with the cache pre-built means there's less potential for downtime during upscale events or new deployments (as those containers shouldn't be available to handle a request until the cache is built, or you have users waiting several seconds for the HTTP response as everything's blocked by the cache being built).

**Reply by eerison** [reply]
Yeah it's make sense,  and I checked the cache generated is small!

Thank you guys for your support ❤️

**Comment by andriusvo** [comment]
@eerison Maybe can you show how you configured to make it work? I'm using Redis as my primary cache service in an application, but want to use Filesystem for Twig. Then I passed service `@twig.cache` it shows me an error `must be of type Twig\Cache\CacheInterface, Symfony\Component\Cache\Adapter\TagAwareAdapter given`. Any solution for that?

**Reply by eerison** [reply]
my solution for this was [cache:warmup](https://bref.sh/docs/frameworks/symfony.html#deploy) during the image build and deploy it. I didn't save into redis ;)

> Then I passed service @twig.cache it shows me an error must be of type Twig\Cache\CacheInterface, Symfony\Component\Cache\Adapter\TagAwareAdapter given. Any solution for that?

it is another issue. But you need to debug why it is passing a different object.

**Reply by andriusvo** [reply]
Yeah, I should debug it for sure, but my question is more if you are passing `@twig.cache` as a service and everything works fine for you?

**Reply by eerison** [reply]
I didn't override my cache handle :/, then I didn't face the issue that you mentioned :/
